src/bun.js/unbounded_queue.zig

```diff
@@ -73,6 +73,8 @@ pub fn UnboundedQueue(comptime T: type, comptime next_field: meta.FieldEnum(T))
         }
 
         pub fn pop(self: *Self) ?*T {
+            if (self.isEmpty()) return null;
```

Comment:
What happens if `count` is incremented after `isEmpty()` returns true?

Why isn't it sufficient for `front` to be null and then return?

Reply:
I think I can revert this, this only evades the problem but is still race, for some reason we are getting an assertion error on `assert(@atomicRmw(usize, &self.count, .Sub, 1, .Monotonic) >= 1);`

![image](https://github.com/oven-sh/bun/assets/6379399/98e9b2e0-72c9-4556-81a3-8a2fc7730819)

Reply:
Maybe we should use `popBatch`